addons/html_editor/static/src/main/toolbar/toolbar_plugin.js

```diff
@@ -341,18 +341,53 @@ export class ToolbarPlugin extends Plugin {
      */
     updateToolbar = debounce(this._updateToolbar, 0, { trailing: true });
     _updateToolbar(selectionData = this.dependencies.selection.getSelectionData()) {
-        const targetedNodes = this.getFilteredTargetedNodes();
-        this.updateNamespace(targetedNodes);
-        this.updateToolbarVisibility(selectionData, targetedNodes);
-        if (!this.overlay.isOpen) {
-            return;
+        // Prevent toolbar to open if the selection is not in the editable area,
+        // or if the selection is protected or protecting.
+        if (
+            !selectionData.currentSelectionIsInEditable ||
+            selectionData.documentSelectionIsProtected ||
+            selectionData.documentSelectionIsProtecting
+        ) {
+            return this.closeToolbar();
```

Comment:
Why does it need to _return_ the call to `closeToolbar`? (besides, you don't need to pass selection data here?)

Reply:
yes please